.changeset/large-spiders-protect.md

```diff
@@ -0,0 +1,15 @@
+---
+"@hashicorp/design-system-components": minor
+---
+
+`Hds::Table`
+- Added `@selectableColumnKey` argument which enables sorting by row selection state and specifies the corresponding selection state key.
+
+`Hds::Table::Tr`
+- Added `@selectableColumnKey` argument which enables sorting by row selection state and specifies the corresponding selection state key.
+- Added `@sortBySelectedOrder` argument which determines the state of the sort button in the selected item column.
+- Added `@onClickSortBySelected` argument which is the callback for the sort button in the selected item column.
+
+`Hds::Table::ThSelectable`
+- Added `@onClickSortBySelected` argument which is the callback for the sort button in the selected item column.
+- Added `@sortBySelectedOrder` argument which determines the state of the sort button in the selected item column.
```

Comment:
thought: some of these are for internal use; I don't think we plan on documenting them all (but I might be wrong) so I suggest we add to the changelog only the arguments we will expose in the component API

Reply:
That was what I had originally, but @didoo suggested that I document all of the additions to subcomponents as well. Curious to hear his thoughts

Reply:
In terms of changelog, we all agree the changes should be mentioned. 
For the website, personally I would document them anyway (and in case add a note clarifying that it's meant for internal use). That said, I totally get that the Table API documentation is already heavy so if you decide differently I would not oppose (but consider that, at times, we use the component API documentation too, to understand what a certain argument does).